### PR TITLE
[plugins/aws][feat] Normalise KMS Key IDs and add edges

### PR DESCRIPTION
**Pull request by anjafr**
# Description

- Add edge between RDS Instances and KMS Keys
- Kms Keys can be identified by a variety of references (Id, Arn, ...). This introduces a normalisation method that leads back to the actual key id, making it simpler to add edges to Keys

# To-Dos

- [x] Add test coverage for new or updated functionality
- [x] Lint and test with `tox`

# Code of Conduct

By submitting this pull request, I agree to follow the [code of conduct](https://resoto.com/code-of-conduct).

### CODE REVIEW

File: plugins/aws/resoto_plugin_aws/resource/rds.py [hunk 1]
```diff
@@ -463,6 +464,21 @@ def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
...
+        if self.rds_kms_key_id:
+            if self.rds_kms_key_id.startswith("arn:"):
+                builder.dependant_node(from_node=self, clazz=AwsKmsKey, arn=self.rds_kms_key_id)
+            else:
+                builder.dependant_node(from_node=self, clazz=AwsKmsKey, id=self.rds_kms_key_id)
+        if self.rds_performance_insights_kms_key_id:
+            if self.rds_performance_insights_kms_key_id.startswith("arn:"):
+                builder.dependant_node(from_node=self, clazz=AwsKmsKey, arn=self.rds_performance_insights_kms_key_id)
+            else:
+                builder.dependant_node(from_node=self, clazz=AwsKmsKey, id=self.rds_performance_insights_kms_key_id)
```
Comment (aquamatthias) [inline plugins/aws/resoto_plugin_aws/resource/rds.py:476]: There is quite some repition.
Maybe we should add a static method in AwsKmsKey which encodes the handling if it is id or arn or something else?

File: plugins/aws/resoto_plugin_aws/resource/rds.py [hunk 2]
```diff
@@ -463,6 +464,17 @@ def connect_in_graph(self, builder: GraphBuilder, source: Json) -> None:
...
+            self.rds_kms_key_id,
+            self.rds_performance_insights_kms_key_id,
+            self.rds_activity_stream_kms_key_id,
+        ]
+        keys = [key for key in potential_keys if key]
+        for key_reference in keys:
+            if key_reference.startswith("arn:"):
+                builder.dependant_node(from_node=self, clazz=AwsKmsKey, arn=key_reference)
+            else:
+                builder.dependant_node(from_node=self, clazz=AwsKmsKey, id=key_reference)
```
Comment (aquamatthias) [inline plugins/aws/resoto_plugin_aws/resource/rds.py:477]: I remember commenting in another PR that we should have a method in AwsKmsKey for that kind of repetition. Do we have it somewhere?

Comment (aquamatthias) [reply plugins/aws/resoto_plugin_aws/resource/rds.py:477]: Oops this PR - am I missing something? I can not see it.

Comment (anjafr) [reply plugins/aws/resoto_plugin_aws/resource/rds.py:477]: @aquamatthias 
Ohh - I had misread that and thought you only meant the repetition we had here for the three Keys in RDS Instances 😅 
The ambivalence of Id or Arn for this edge (and a few others like it) doesn't lie within `AwsKmsKey` though, so I don't see how we could resolve it over there? It lies within the response of e.g. `rds describe-db-instances` 🤔

Comment (anjafr) [reply plugins/aws/resoto_plugin_aws/resource/rds.py]: @aquamatthias Here we go now!

**Pull request merged**